src/lib/receiver.js

```diff
@@ -281,44 +289,48 @@ function createReceiver (opts) {
-
-        /**
-         * [IncomingTransfer](https://github.com/interledger/rfcs/blob/master/0004-ledger-plugin-interface/0004-ledger-plugin-interface.md#incomingtransfer) from the ledger plugin and the fulfillment string
-         *
-         * @event incoming
-         * @type {object}
-         */
-        eventEmitter.emit('incoming', transfer, fulfillment)
-
-        if (protocol === 'ipr') {
-          /**
-           * [IncomingTransfer](https://github.com/interledger/rfcs/blob/master/0004-ledger-plugin-interface/0004-ledger-plugin-interface.md#incomingtransfer) from the ledger plugin and the fulfillment string for a specific request
-           *
-           * @event incoming:<requestid>
-           * @type {object}
-           */
-          // Allow listeners for specific requests and on wildcard events such that
-          // `incoming:appid.*` will match `incoming:appid:requestid`
-          eventEmitter.emit('incoming:' + requestId, transfer, fulfillment)
-        } else if (protocol === 'ssp') {
-          /**
-           * [IncomingTransfer](https://github.com/interledger/rfcs/blob/master/0004-ledger-plugin-interface/0004-ledger-plugin-interface.md#incomingtransfer) from the ledger plugin and the fulfillment string for a specific token
-           *
-           * @event incoming:ssp:<token>
-           * @type {object}
-           */
-          eventEmitter.emit('incoming:ssp:' + requestId, transfer, fulfillment)
-        }
-
-        return 'sent'
-      })
-      .catch((err) => {
-        debug('error submitting fulfillment', err)
+        debug('about to submit fulfillment: ' + fulfillment)
+        return client.fulfillCondition(transfer.id, fulfillment)
+          .then(() => {
+            debug('successfully submitted fulfillment ' + fulfillment + ' for request ' + requestId + ' (transfer ' + transfer.id + ')')
+
+            /**
+            * [IncomingTransfer](https://github.com/interledger/rfcs/blob/master/0004-ledger-plugin-interface/0004-ledger-plugin-interface.md#incomingtransfer) from the ledger plugin and the fulfillment string
+            *
+            * @event incoming
+            * @type {object}
+            */
+            eventEmitter.emit('incoming', transfer, fulfillment)
+
+            if (protocol === 'ipr') {
+              /**
+              * [IncomingTransfer](https://github.com/interledger/rfcs/blob/master/0004-ledger-plugin-interface/0004-ledger-plugin-interface.md#incomingtransfer) from the ledger plugin and the fulfillment string for a specific request
+              *
+              * @event incoming:<requestid>
+              * @type {object}
+              */
+              // Allow listeners for specific requests and on wildcard events such that
+              // `incoming:appid.*` will match `incoming:appid:requestid`
+              eventEmitter.emit('incoming:' + requestId, transfer, fulfillment)
```

Comment:
Looking at this now... I doubt we'll break a lot of code if we change this to `incoming:ipr:`, so maybe we should just rip off the band-aid. :)